### ISSUE TITLE
Warning that termination is not guaranteed

### ISSUE THREAD

**Issue by MarkTiedemann** [sindresorhus/taskkill#6]
I think it would be good if the Readme warned people about false positives.

Even if `SUCCESS: Sent termination signal to the process "abc.exe" with PID 123` is returned, there is no guarantee that the process will actually be terminated (see example below) - it's just sending the signal, not more, not less.

```
λ tasklist /fi "imageName eq boinc.exe"                                     
                                                                            
Image Name                     PID Session Name        Session#    Mem Usage
========================= ======== ================ =========== ============
boinc.exe                    12312 Console                    1     14,108 K
                                                                            
λ taskkill /im "boinc.exe"                                                  
SUCCESS: Sent termination signal to the process "boinc.exe" with PID 12312. 
                                                                            
λ tasklist /fi "imageName eq boinc.exe"                                     
                                                                            
Image Name                     PID Session Name        Session#    Mem Usage
========================= ======== ================ =========== ============
boinc.exe                    12312 Console                    1     16,616 K
                                                                            
λ taskkill /im "boinc.exe" /f                                               
SUCCESS: The process "boinc.exe" with PID 12312 has been terminated.        
                                                                            
λ tasklist /fi "imageName eq boinc.exe"                                     
INFO: No tasks are running which match the specified criteria.              
```

Just to be clear: the warning should apply to the `force` option as well: for example, if you are trying to kill a task that depends on a service, you will have to kill the service first.

There are also the usual suspects, of course: insufficient permissions, unkillable system processes, faulty driver processes, etc. All in all, you should probably always check whether the process you tried to kill was actually killed.

PS: Just in case anyone wants to parse the `taskkill` output to report info, success, errors, etc. in a more user friendly way: please note that these messages are i18n'ed (for example, in English: `SUCCESS: ...`, in German: `ERFOLGREICH: ...`).

PPS: It might be worth investigating what the pros and cons of using `taskkill` compared to [`wmic`](https://msdn.microsoft.com/en-us/library/aa394531(v=vs.85).aspx) or [`Stop-Process`](https://msdn.microsoft.com/en-us/powershell/reference/5.1/microsoft.powershell.management/stop-process) are. Perhaps there's a better way to kill stuff on Windows ... I only know for a fact that `shutdown /t 0 /f /r` is very reliable when it comes to killing processes. Unfortunately, however, it has terrible performance. :/

**Comment by sindresorhus** [comment]
Good points. Maybe we could use [`tasklist`](https://github.com/sindresorhus/tasklist) to check if the process was killed or not?

**Comment by MarkTiedemann** [comment]
@sindresorhus I guess as a user of this module you have already shown intent that you want to kill the task, and thus are also interested in knowing whether it was actually killed or not.

I'd rather not check for it in this module, however. I think it's out of scope: If we do check termination, this module wouldn't be a low-level wrapper around `taskkill` anymore. It would suddenly be a higher-level API which wraps around both `taskkill` and, indirectly, around `tasklist`. So from this viewpoint of modularity, I think it makes sense to create a third higher-level module, perhaps called `win-kill`, that combines both the `taskkill` and the `tasklist` module into a nice, user-friendly API.

Your thoughts? :)

**Comment by sindresorhus** [comment]
Agreed. I think using [`fkill`](https://github.com/sindresorhus/fkill) as such abstraction method would be better. I don't see much value in having an abstraction like this for each OS. Better just have it in `fkill` cross-platform.

**Comment by MarkTiedemann** [comment]
Yeah, I agree. But `fkill` doesn't check for termination at the moment, right?

So based on our discussion, I think there should be an option for `fkill`, such as, `checkTermination` (or perhaps `confirmKill`?) with a default value of `true`.

If I have time tomorrow, I might send 2 PRs: one to update the `taskkill` readme with a warning, and another one for `fkill` to implement the check option.

Have a good day! :)

**Comment by sindresorhus** [comment]
> Yeah, I agree. But fkill doesn't check for termination at the moment, right?

Not at the moment, but I think that would be useful.

> (or perhaps confirmKill?) with a default value of true.

👍 

> I might send 2 PRs: one to update the taskkill readme with a warning, and another one for fkill to implement the check option.

🌈

> Have a good day! :)

You too! (Yes, I realize I'm a week late 😝)
